### PR TITLE
Adding retry functionality when parallel is set to True

### LINKED ISSUE #61630: [BUG] retry state option unavailble when parallel is True
**Description**
When the `retry` options are set in a state and parallel is set to True, the job is rerun regardless of the original state results and the subsequent runs are not done in parallel.

**Setup**
Basic state that includes `parallel: True` and `retry` options:

```
{% set parallel = True %}
{% for i in range(1,5) %}
run {{ i }}:
  cmd.run:
    - name: /tmp/cmd.sh
    - parallel: {{ parallel }}
    - retry:
        attempts: 2
        until: True
        interval: 5
{% endfor %}
```

```
randomNumber=$(( $RANDOM % 2 ))

if [[ $randomNumber == 1 ]]; then
    echo "False"
    exit 1
else
    echo "True"
    exit 0
fi
```

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD

```
PASTE HERE
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Making the retry state system feature available when parallel is set to True.

### What issues does this PR fix or reference?
Fixes: #61630 #61439

### Previous Behavior
When the retry options are set in a state and parallel is set to True, the job is rerun regardless of the original state results and the subsequent runs are not done in parallel.

### New Behavior
Retry functionality is properly honored with `parallel: True`

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/state.py [hunk 1]
```diff
@@ -2016,11 +2016,74 @@ def _call_parallel_target(self, name, cdata, low):
...
+
+                    interval = low["retry"]["interval"]
+                    if low["retry"]["splay"] != 0:
+                        interval = interval + random.randint(0, low["retry"]["splay"])
+                    log.info(
+                        "State result does not match retry until value, "
+                        "state will be re-run in %s seconds",
+                        interval,
+                    )
+                    self.functions["test.sleep"](interval)
```
Comment (Ch3LL) [inline salt/state.py:2045]: Why are we calling the `test.sleep` function here instead of just using python's time's lib with `time.sleep`?

**Review by Ch3LL — COMMENTED** [review]
just a question

File: salt/state.py (hunk 1 above)
Comment (garethgreenaway) [reply salt/state.py:2045]: No particular reason, most of this code was copied from the existing `retry` functionality.  In this instance it probably makes more sense to use `time.sleep` as this is running inside it's own thread/process.  I'll update it.

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
Looks like the windows test failures are related to this PR

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
is this a related failure? https://jenkins.saltproject.io/job/pr-macosx-catalina-x86_64-py3-pytest/job/PR-61631/22/testReport/

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**